tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AWS/Lambda/LambdaRequestBuilder.cs

```diff
@@ -35,12 +35,17 @@ WebRequest ILambdaExtensionRequest.GetStartInvocationRequest()
         return request;
     }
 
-    WebRequest ILambdaExtensionRequest.GetEndInvocationRequest(Scope scope, bool isError)
+    WebRequest ILambdaExtensionRequest.GetEndInvocationRequest(Scope scope, object state, bool isError)
     {
         var request = WebRequest.Create(Uri + EndInvocationPath);
         request.Method = "POST";
         request.Headers.Set(HttpHeaderNames.TracingEnabled, "false");
 
+        if (state != null)
+        {
+            request.Headers.Set("lambda-runtime-aws-request-id", (string)state);
+        }
```

Comment:
Although it should never happen, if `state` is ever a different type from `string`, this casting operation will throw an exception. It would be safer to check that `state` is a `string`, not simply not null

The more modern way of doing this in C# is by combining the type check with a `string` variable declaration and removing the `(string)` casting operator:
```csharp
if (state is string requestId)
{
    request.Headers.Set("lambda-runtime-aws-request-id", requestId);
}